torchvtk/augmentation/DictTransform.py

```diff
@@ -0,0 +1,155 @@
+
+            # to GPU.
+            if self.device == "cuda":
+                tmp = tmp.to(self.device)
+
+            if tmp.dtype is torch.float16:
+                tmp = tmp.to(torch.float32)
+
+            tmp = self.transform(tmp)
+
+            if self.dtype is torch.float16:
+                tmp = tmp.to(torch.float16)
+            elif self.dtype is not torch.float16 or torch.float32:
+                tmp = tmp.to(torch.float32)
+            tmp = tmp.to(self.device)
+            data[key] = tmp
+        return data
+
+
+class NoiseDictTransform(DictTransform):
+    """
+    Transformations for adding noise to images.
+    """
+
+    def __init__(self, noise_variance=(0.001, 0.05), **kwargs):
+        """
+
+        :param noise_variance: The variance of the noise added to the  image.
+        :param kwargs: Arguments of the super class.
+        """
+        self.device = kwargs["device"]
+        self.noise_variance = noise_variance
+        DictTransform.__init__(self, **kwargs)
+
+    def transform(self, data):
+        std = torch.rand(data.size(0) if data.ndim > 5 else 1, device=data.device, dtype=data.dtype)
+        return data + torch.randn_like(data) * std
+
+
+class BlurDictTransform(DictTransform):
+    """Transformation for adding Blur to images."""
+
+    def __init__(self, channels=1, kernel_size=(3, 3, 3), sigma=1, **kwargs):
+        """
+        :param channels: Amount of channels of the input data.
+        :param kernel_size: Size of the convolution kernel.
+        :param sigma: Standard deviation.
+        """
+        self.channels = channels
+        self.sigma = [sigma, sigma, sigma]
+        self.kernel_size = kernel_size
+        self.device = kwargs["device"]
+        DictTransform.__init__(self, **kwargs)
+        # code from: https://discuss.pytorch.org/t/is-there-anyway-to-do-gaussian-filtering-for-an-image-2d-3d-in-pytorch/12351/10
+        # initialize conv layer.
+        kernel = 1
+        # todo add dynamic padding for higher kerner_size
+        #
+        self.meshgrids = torch.meshgrid(
+            [
+                torch.arange(size, dtype=torch.float32)
+                for size in self.kernel_size
+            ]
+        )
+        for size, std, mgrid in zip(self.kernel_size, self.sigma, self.meshgrids):
+            mean = (size - 1) / 2
+            kernel *= 1 / (std * math.sqrt(2 * math.pi)) * torch.exp(-((mgrid - mean) / std) ** 2 / 2)
+
+        # Make sure sum of values in gaussian kernel equals 1.
+        kernel = kernel / torch.sum(kernel)
+
+        # Reshape to depthwise convolutional weight
+        kernel = kernel.view(1, 1, *kernel.size())
+        kernel = kernel.repeat(self.channels, *[1] * (kernel.dim() - 1))
+
+        kernel = kernel.to(self.device)
+        self.weight = kernel
+        self.conv = F.conv3d
+
+    def transform(self, data):
+        vol = make_5d(data)
+        vol = self.conv(vol, weight=self.weight, groups=self.channels, padding=1)
```

Comment:
For blurring we would need to determine the amount using `padding = kernel_size // 2`.
Also usually replication padding is used when doing a gaussian blur, since zero padding creates a gradient at the borders. Assuming `self.pad = kernel_size // 2` I would suggest
```suggestion
        vol = F.pad(make_5d(data), (self.pad, )*6, mode='replicate')
        return self.conv(vol, weight=self.weight, groups=self.channels, padding=0)
```